app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -25,7 +25,8 @@ class Api::Rhsm::CandlepinProxiesController < Api::V2::ApiController
 
     before_action :check_content_type, :except => :async_hypervisors_update
 
-    skip_before_action :set_taxonomy, :only => [:rhsm_index, :consumer_activate, :consumer_create]
```

Comment:
this worked but user authentication (which is triggered when subscription-manager is used with username and password) explicitly calls `set_taxonomy` which is why it tried to set the taxonomy for these actions

anyway, using set_taxonomy is right, so let's use it, see lines below